include/picolibrary/crc.h

```diff
@@ -590,6 +590,171 @@ class Augmented_Nibble_Lookup_Table_Calculator {
     }
 };
 
+/**
+ * \brief Direct nibble lookup table calculator.
+ *
+ * \tparam Register_Type Calculation register type.
+ */
+template<typename Register_Type>
+class Direct_Nibble_Lookup_Table_Calculator {
+  public:
+    /**
+     * \brief Calculation register type.
+     */
+    using Register = Register_Type;
+
+    /**
+     * \brief Constructor.
+     */
+    constexpr Direct_Nibble_Lookup_Table_Calculator() noexcept = default;
+
+    /**
+     * \brief Constructor.
+     *
+     * \param[in] parameters The calculation parameters.
+     */
+    constexpr explicit Direct_Nibble_Lookup_Table_Calculator( Parameters<Register> const & parameters ) noexcept :
+        m_lookup_table{ generate_nibble_lookup_table( parameters.polynomial ) },
+        m_preprocessed_initial_remainder{
+            preprocess_initial_remainder( parameters.initial_remainder, m_lookup_table )
+        },
+        m_process_input{ input_processor( parameters.input_is_reflected ) },
+        m_process_output{ output_processor<Register>( parameters.output_is_reflected ) },
+        m_xor_output{ parameters.xor_output }
+    {
+    }
+
+    /**
+     * \brief Constructor.
+     *
+     * \param[in] source The source of the move.
+     */
+    constexpr Direct_Nibble_Lookup_Table_Calculator( Direct_Nibble_Lookup_Table_Calculator && source ) noexcept = default;
+
+    /**
+     * \brief Constructor.
+     *
+     * \param[in] original The original to copy.
+     */
+    constexpr Direct_Nibble_Lookup_Table_Calculator(
+        Direct_Nibble_Lookup_Table_Calculator const & original ) noexcept = default;
+
+    /**
+     * \brief Destructor.
+     */
+    ~Direct_Nibble_Lookup_Table_Calculator() noexcept = default;
+
+    /**
+     * \brief Assignment operator.
+     *
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Nibble_Lookup_Table_Calculator && expression ) noexcept
+        -> Direct_Nibble_Lookup_Table_Calculator & = default;
+
+    /**
+     * \brief Assignment operator.
+     *
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    constexpr auto operator=( Direct_Nibble_Lookup_Table_Calculator const & expression ) noexcept
+        -> Direct_Nibble_Lookup_Table_Calculator & = default;
+
+    /**
+     * \copydoc picolibrary::CRC::Calculator_Concept::calculate()
+     */
+    template<typename Iterator>
+    auto calculate( Iterator begin, Iterator end ) const noexcept -> Register
+    {
+        // #lizard forgives the length
+
+        constexpr auto nibble_digits = std::numeric_limits<std::uint8_t>::digits / 2;
+
+        auto remainder = m_preprocessed_initial_remainder;
+
+        for ( ; begin != end; ++begin ) {
+            auto const processed_input = ( *m_process_input )( *begin );
+
+            auto const nibbles = Fixed_Size_Array<std::uint_fast8_t, 2>{
```

Comment:
Generalize nibble types and operations:
- `::picolibrary::CRC::Nibble` (`std::uint_fast8_t`)
- `::picolibrary::CRC::NIBBLE_DIGITS`
- `::picolibrary::CRC::Nibbles` (`::picolibrary::Fixed_Size_Array<::picolibrary::CRC::Nibble, 2>`)
- `::picolibrary::CRC::byte_as_nibbles()` (convert `std::uint8_t` to `::picolibrary::CRC::Nibbles`)

It may be possible to remove `// lizard forgives the length` once this is done.